### ISSUE TITLE
IRIError and SchemaError are not used

### ISSUE THREAD

**Issue by robstewart57** [robstewart57/rdf4h#57]
ping @wismill 

Where are `IRIError` and `SchemaError` used?

I would've thought that the left value for the following functions:

```
mkIRI :: Text -> Either String IRI
parseIRI :: Text -> Either String IRIRef
parseRelIRI :: Text -> Either String IRIRef
validateIRI :: Text -> Either String Text
resolveIRI :: Text -> Text -> Either String Text
```

Would be `IRIError` or `SchemaError` rather than `String`?

**Comment by wismill** [comment]
You are right, they are not used yet. The compiler did not catch this because they are in the export list.

Maybe we could keep things simple and use the parser combinator `<?>` to make the error message and remove `IRIError` and `SchemaError`?

**Comment by robstewart57** [comment]
> Maybe we could keep things simple and use the parser combinator <?> to make the error message and remove IRIError and SchemaError?

Sounds sensible to keep things simple.

How about using the library's existing `ParseFailure`: https://github.com/robstewart57/rdf4h/blob/8b5b06fb9ec59d0370e301ddd444b6e3291cf226/src/Data/RDF/Types.hs#L483

E.g. it's used https://github.com/robstewart57/rdf4h/blob/8b5b06fb9ec59d0370e301ddd444b6e3291cf226/src/Data/RDF/Types.hs#L403

 in

```
class RdfParser p where
  parseString :: (Rdf a) => p -> Text -> Either ParseFailure (RDF a)
  parseFile :: (Rdf a) => p -> String -> IO (Either ParseFailure (RDF a))
  parseURL :: (Rdf a) => p -> String -> IO (Either ParseFailure (RDF a))
```

I.e.

```
mkIRI :: Text -> Either ParseFailure IRI
parseIRI :: Text -> Either ParseFailure IRIRef
parseRelIRI :: Text -> Either ParseFailure IRIRef
validateIRI :: Text -> Either ParseFailure Text
resolveIRI :: Text -> Text -> Either ParseFailure Text
```
